walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,92 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendees = request.getAttendees();
+    long duration = request.getDuration();
+
+    //Case in which the requested meeting has no attendies
+    if(attendees.isEmpty()){
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    //Case in which the requested meeting has a duration that is too long/short
+    if(duration > TimeRange.WHOLE_DAY.duration() || duration < 0){
+        return Arrays.asList();
+    } 
+
+    //List of time ranges of other events that attendees are attending
+    List<TimeRange> timeRanges = new ArrayList<TimeRange>();
```

Comment:
Consider using set here because we want to avoid dupes.

Reply:
I used a set at first but then switched it to an ArrayList so I could sort it in ascending order based on the start time of the TimeRange (see line 57).  This rest of the algorithm is based off of the list timeRanges being in the correct order, as it starts at the beginning of the day and iterates through this list to chronologically see when people are available. 
I just did some research though and found that there is a SortedSet.  I don't know much about this type of set... do you think it would be a good solution?

Reply:
You can use TreeSet as SortedSet. Or you could simply use hashSet to find all the timeRanges and add them all to a arrayList.